constraint/pkg/instrumentation/instrumentation.go

```diff
@@ -0,0 +1,24 @@
+// package instrumentation defines primitives to
+// support more observability throughout the contraint framework.
+package instrumentation
+
+// NamedStats covers primitive for many heterogenoues statstics and metrics
+// to be represented to consumers.
+type NamedStats struct {
+	StatsName string
+	Stats     []*Stat
+}
+
+// Stat is a Name, Value, Description tuple that may contain
+// metrics or aggregations of metrics.
+type Stat struct {
+	Name        string
+	Value       string
```

Comment:
Will Value always be string?

Reply:
i was thinking so. At first, I thought we could have it as a more generic type but decided to start w string since consumers are free to take on the conversion

Reply:
Conversion in the sense of parsing strings can be expensive and error-prone. If we want to preserve the potential for strong types, some kind of `interface{}` value is likely a better fit (maybe require it supports a String() method? Worth looking at what would be necessary to make things easy to JSON-ify).